pkg/controller/kibana/config_settings.go

```diff
@@ -263,8 +274,33 @@ func elasticsearchTLSSettings(kb kbv1.Kibana) map[string]interface{} {
 // esCaCertSecretVolume returns a SecretVolume to hold the Elasticsearch CA certs for the given Kibana resource.
 func esCaCertSecretVolume(kb kbv1.Kibana) volume.SecretVolume {
 	return volume.NewSecretVolumeWithMountPath(
-		kb.AssociationConf().GetCASecretName(),
+		kb.EsAssociation().AssociationConf().GetCASecretName(),
 		"elasticsearch-certs",
 		esCertsVolumeMountPath,
 	)
 }
+
+// entCaCertSecretVolume returns a SecretVolume to hold the EnterpriseSearch CA certs for the given Kibana resource.
+func entCaCertSecretVolume(kb kbv1.Kibana) volume.SecretVolume {
+	return volume.NewSecretVolumeWithMountPath(
+		kb.EntAssociation().AssociationConf().GetCASecretName(),
+		"ent-certs",
+		entCertsVolumeMountPath,
+	)
+}
+
+func enterpriseSearchSettings(kb kbv1.Kibana) map[string]interface{} {
+	cfg := map[string]interface{}{}
+	assocConf := kb.EntAssociation().AssociationConf()
+	if assocConf.URLIsConfigured() {
+		cfg[EnterpriseSearchHost] = assocConf.GetURL()
+	}
+	if assocConf.CAIsConfigured() {
+		cfg[EnterpriseSearchSslCertificateAuthorities] = filepath.Join(entCertsVolumeMountPath, certificates.CAFileName)
+		// Rely on "certificate" verification mode rather than "full" to allow Kibana
+		// to connect to Enterprise Search through the k8s-internal service DNS name
+		// even though the user-provided certificate may only specify a public-facing DNS.
+		cfg[EnterpriseSearchSslVerificationMode] = "certificate"
```

Comment:
@kovyrin fyi this is what we also do for the Kibana -> Elasticsearch connection.
Less secure but more convenient when using internal k8s dns services.
Users are free to override that configuration option through the Kibana CRD if they want to. Users setting will take precedence.
Let me know if you think that's a problem!

Reply:
I think it makes sense to keep this in line with what we do for other products. It is definitely a lot more convenient and makes sense as a default in a world where TLS certificate names are pretty arbitrary (depend on resource names).